### PR TITLE
Display trusted auth extensions

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
New look for this quick pick.
image

Now if you try to un-check one of the trusted ones, it won't check.

Also, the product.json entry takes a new shape... specifying which extensions use which auth providers.

### CODE REVIEW

**Review by DonJayamanne — DISMISSED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Comment by gjsjohnmurray** [comment]
Imagine the improvement to readability if #75835 landed 😉

**Comment by gjsjohnmurray** [comment]
Also, I suggest the tip on "Trusted by Microsoft" should indicate that these items intentionally cannot be unchecked. Otherwise I foresee bug reports.

image

Maybe rephrase to become "always has access to this account". Even better, add a link to a doc page/section explaining why.

Footnote: I was interested to note that the Git extension hasn't used the account even though I've performed operations with it.

**Comment by gjsjohnmurray** [comment]
Another idea for improvement is to add a button to each item which would take me to the extension details (e.g. by using `command:workbench.extensions.search` with an `@id:xx.yy` query).

**Comment by TylerLeonhardt** [comment]
Did the word changes. Let me capture the button idea in another issue.
